lib/jekyll/url.rb

```diff
@@ -84,17 +84,36 @@ def generate_url_from_hash(template)
       end
     end
 
+    # We include underscores in keys to allow for 'i_month' and so forth.
+    # This poses a problem for keys which are followed by and underscore
+    # but the underscore is not part of the key, e.g. '/:month_:day'.
+    # That should be :month and :day, but our key extraction regexp isn't
+    # smart enough to know that so we have to make it an explicit
```

Comment:
**personal nit:** it would be better to have a `,` (comma) between `..that` and `so..` :smiley: